src/sql/arrow_sql_gen/statement.rs

```diff
@@ -483,6 +483,70 @@ impl InsertBuilder {
+                        IntervalUnit::DayTime => {
+                            let array = column
+                                .as_any()
+                                .downcast_ref::<array::IntervalDayTimeArray>();
+
+                            if let Some(valid_array) = array {
+                                if valid_array.is_null(row) {
+                                    row_values.push(Keyword::Null.into());
+                                    continue;
+                                }
+
+                                let days = valid_array.value(row).days;
+                                let milliseconds = valid_array.value(row).milliseconds;
+
+                                let interval_str =
+                                    format!("{days} days {milliseconds} milliseconds");
+
+                                row_values.push(interval_str.into());
+                            }
+                        }
+                        IntervalUnit::YearMonth => {
+                            let array = column
+                                .as_any()
+                                .downcast_ref::<array::IntervalYearMonthArray>();
+
+                            if let Some(valid_array) = array {
+                                if valid_array.is_null(row) {
+                                    row_values.push(Keyword::Null.into());
+                                    continue;
+                                }
+
+                                let months = valid_array.value(row);
+
+                                let interval_str = format!("{months} months");
+
+                                row_values.push(interval_str.into());
+                            }
+                        }
+                        // The smallest unit in Postgres of inetrval is microsecond
+                        // Cast with loss of precision in nano second
+                        IntervalUnit::MonthDayNano => {
+                            let array = column
+                                .as_any()
+                                .downcast_ref::<array::IntervalMonthDayNanoArray>();
+
+                            if let Some(valid_array) = array {
+                                if valid_array.is_null(row) {
+                                    row_values.push(Keyword::Null.into());
+                                    continue;
+                                }
+
+                                let months = valid_array.value(row).months;
+                                let days = valid_array.value(row).days;
+                                let nanoseconds = valid_array.value(row).nanoseconds;
+                                let micros = nanoseconds / 1_000;
+
+                                let interval_str =
+                                    format!("{months} months {days} days {micros} microseconds");
+
+                                row_values.push(interval_str.into());
+                            }
```

Comment:
some of above might be able to use arrow's `array_value_to_string` function.

Reply:
Thanks for pointing out! I updated to use array_value_to_string for arrow - row conversion, and use the manual parsing as a fallback when array_value_to_string returns error